sunbird-es-utils/src/main/java/org/sunbird/common/inf/ElasticSearchUtil.java

```diff
@@ -0,0 +1,114 @@
+package org.sunbird.common.inf;
+
+import java.util.List;
+import java.util.Map;
+import org.sunbird.common.models.response.Response;
+import org.sunbird.common.models.util.ProjectUtil.EsType;
+import org.sunbird.dto.SearchDTO;
+import scala.concurrent.Future;
+
+public interface ElasticSearchUtil {
+  /**
+   * This method will put a new data entry inside Elastic search. identifier value becomes _id
+   * inside ES, so every time provide a unique value while saving it.
+   *
+   * @param index String ES index name
+   * @param type String ES type name
+   * @param identifier ES column identifier as an String
+   * @param data Map<String,Object>
+   * @return String identifier for created data
+   */
+  public Future<String> createData(
+      String index, String type, String identifier, Map<String, Object> data);
+
+  /**
+   * This method will update data based on identifier.take the data based on identifier and merge
+   * with incoming data then update it.
+   *
+   * @param index String
+   * @param type String
+   * @param identifier String
+   * @param data Map<String,Object>
+   * @return boolean
+   */
+  public Future<Boolean> updateData(
+      String index, String type, String identifier, Map<String, Object> data);
+
+  /**
+   * This method will provide data form ES based on incoming identifier. we can get data by passing
+   * index and identifier values , or all the three
+   *
+   * @param type String
+   * @param identifier String
+   * @return Map<String,Object> or null
+   */
+  public Future<Map<String, Object>> getDataByIdentifier(
+      String index, String type, String identifier);
+
+  /**
+   * This method will remove data from ES based on identifier.
+   *
+   * @param index String
+   * @param type String
+   * @param identifier String
+   */
+  public Future<Boolean> removeData(String index, String type, String identifier);
+
+  /**
+   * This method will do the data search inside ES. based on incoming search data.
+   *
+   * @param index String
+   * @param type String
+   * @param searchData Map<String,Object>
+   * @return Map<String,Object>
+   */
+  public Future<Map<String, Object>> searchData(
```

Comment:
there are two method:
searchData
and 
complexSearch 
are we using both method in code or only one is used 
if only one is used then will delete unused on and make method name as search only.

Reply:
Remove search method and changed complex search -> search. we were to suing search method .